grails-app/assets/javascripts/controllers/modal-video-ctrl.js

```diff
@@ -1,22 +1,49 @@
 'use strict';
 
+/** Directive for formatting from mm:ss to int and back */
+streamaApp.directive("formatDirective", function(){
```

Comment:
Several Points:
- Directives should be placed in their own file and not inside of a controller file.
- directives should not be suffixed with the word "directive" but instead prefixed with an app-specific token such as "streama" or something else. In this case I would call it videoTimeFormat or streamaVideoTimeFormat and then it is used as video-time-format. Just "format-directive" is bad practice and too generic. 

Otherwise good joob on this.